### PR TITLE
#137 Fix build and update readme.

### PR DESCRIPTION
**Pull request by tortmayr**
Resolves eclipse-glsp/glsp/issues/137

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,6 +1,29 @@
...
+# Eclipse GLSP Examples ![build-status](https://img.shields.io/jenkins/build?jobUrl=https%3A%2F%2Fci.eclipse.org%2Fglsp%2Fjob%2Feclipse-glsp%2Fjob%2Fglsp-examples%2Fjob%2Fmaster%2F)
+
+This repository contains code examples that demonstrates how build diagram editors with the [Graphical Language Server Platform (GLSP)](https://github.com/eclipse-glsp/glsp). The examples are focused on the integration of GLSP editors with the cloud-based [Eclipse Theia IDE](https://github.com/theia-ide/theia) using the [GLSP theia integration](https://github.com/eclipse-glsp/glsp-theia-integration).
```
Comment (martin-fleck-at) [inline README.md:3]: Typos:
- "[...] code examples that demonstrates [...]" -> "[...] code examples that demonstrate [...]" (no 's' on demonstrates)
- "[...] demonstrate how build diagram editors [...]" -> "[...] demonstrate how to build diagram editors [...]" (missing 'to')
- "[...] using the GLSP theia integration." -> "[...] using the GLSP Theia integration." (capitalize 'Theia')

File: README.md [hunk 2]
```diff
@@ -1,6 +1,29 @@
...
+
+Each examples is self-contained and provides both, a example diagram client and its corresponding GLSP server.
```
Comment (martin-fleck-at) [inline README.md:5]: Typos:
- "Each examples [...]" -> "Each example [...]" (no 's')
- "[...] provides both, a example diagram client [...]" -> "[...] provides both, an example diagram client [...]" (missing 'n')

File: README.md [hunk 3]
```diff
@@ -1,6 +1,29 @@
...
+The web-based/client part of the examples has been developed using [Visual Studio Code](https://code.visualstudio.com/) and the server/java part has been developed with the [Eclipse IDE](https://www.eclipse.org/ide/). However, it's of course also possible to use any other IDE or text editor.
+
+## Usage
+- `git clone https://github.com/eclipse-glsp/glsp-examples.git`
+- `cd <any-example-folder>`
+- `cd client && yarn && cd ..` (build client)
+- `cd server && mvn clean install && cd ..` (build server)
+
+## Structure
+- [Minimal Example](https://github.com/eclipse-glsp/glsp-examples/tree/master/minimal): A very simple GLSP Editor for rectangular nodes. This demonstrates the core concepts and basic client-server integration into Theia.
```
Comment (martin-fleck-at) [inline README.md:25]: Typo: "[...] GLSP Editor for rectangular nodes [...]" -> "[...] GLSP editor for rectangular nodes [...]" (previously 'editor' was always written in lower case, consistent usage)

File: Jenkinsfile [hunk 4]
```diff
@@ -81,14 +81,5 @@ pipeline {
                 }
             }
         }
-        
-
-         stage('Deploy client & server (master only)') {
-            when { branch 'master'}
-            steps {
-                 build job: 'deploy-npm-glsp-examples', wait: false
```
Comment (martin-fleck-at) [inline Jenkinsfile:89]: I am not clear on what the reasoning is to remove the build jobs do we not need them anymore? Could you please provide some details on that?

File: README.md [hunk 5]
```diff
@@ -1,6 +1,29 @@
...
+- [Java](https://www.oracle.com/java/technologies/javase-jdk11-downloads.html) `>=11`
+- [Maven](https://maven.apache.org/) `>=3.6.0`
+
+ The examples are heavily interweaved with Eclipse Theia, so please also check the [prerequisites of Theia](https://github.com/eclipse-theia/theia/blob/master/doc/Developing.md#prerequisites).
+
+The web-based/client part of the examples has been developed using [Visual Studio Code](https://code.visualstudio.com/) and the server/java part has been developed with the [Eclipse IDE](https://www.eclipse.org/ide/). However, it's of course also possible to use any other IDE or text editor.
+
+## Usage
+- `git clone https://github.com/eclipse-glsp/glsp-examples.git`
+- `cd <any-example-folder>`
```
Comment (martin-fleck-at) [inline README.md:20]: It says 'any example folder' but right now all I can see is the 'minimal', is there supposed to be more? Or is there supposed to be more soon? If not I would suggest to simply mention the 'minimal' directly or at least add it as an example, e.g., ` (for example "minimal")` as this should always be available.

File: README.md [hunk 6]
```diff
@@ -1,6 +1,29 @@
...
+- `cd client && yarn && cd ..` (build client)
+- `cd server && mvn clean install && cd ..` (build server)
```
Comment (martin-fleck-at) [inline README.md:22]: This step is not correct. When I perform it I get
```
[ERROR] The goal you specified requires a project to execute but there is no POM in this directory (/home/martin/Temp/GLSP/glsp-examples/minimal/server). Please verify you invoked Maven from the correct directory. -> [Help 1]
```
and indeed there is no pom.xml in the server directory so I probably need to go into `org.eclipse.glsp.example.minimal` inside the server directory. Build there provides me (among some other things the following files:
- org.eclipse.glsp.example.minimal-0.0.1-SNAPSHOT-glsp.jar
- org.eclipse.glsp.example.minimal-0.0.1-SNAPSHOT-sources.jar
- org.eclipse.glsp.example.minimal-0.0.1-SNAPSHOT.jar

While it is obvious to me what the SNAPSHOT and SNAPSHOT-sources is, what is the SNAPSHOT-glsp.jar?

Maybe, if you want to keep the generic 'Usage' section as is, the additional information could go into the "Minimal Example" section below?

File: README.md [hunk 7]
```diff
@@ -1,6 +1,29 @@
...
+
```
Comment (martin-fleck-at) [inline README.md:23]: Are there some steps missing? If I perform all the steps as described here everything is built but nothing is running and I cannot see anything, so I would not exactly call that "Usage". I assume that we need to start the server and the client somehow. Could you maybe add these steps as well?

From what I understand through trial and error I need to start the server as follows:
- `cd server/org.eclipse.glsp.example.minimal/target && java -jar org.eclipse.glsp.example.minimal-0.0.1-SNAPSHOT-glsp.jar`

and the client like this:
- `cd client/browser-app && yarn start`

but even then it does not work for me. I always get these errors:
image

In the output of the server it seems that the client is not trying to connect to it at all.

File: README.md [hunk 8]
```diff
@@ -1,6 +1,29 @@
...
+## Structure
```
Comment (martin-fleck-at) [inline README.md:24]: Not sure if "Structure" is the best headline here, maybe just "Examples"?

File: README.md [hunk 9]
```diff
@@ -1,6 +1,29 @@
...
+- [Minimal Example](https://github.com/eclipse-glsp/glsp-examples/tree/master/minimal): A very simple GLSP Editor for rectangular nodes. This demonstrates the core concepts and basic client-server integration into Theia.
 
-An example diagram client and server that has been built with the [Graphical Language Server Platform (GLSP)](https://github.com/eclipse-glsp/glsp). The example editor has been integrated with the cloud-based [Eclipse Theia IDE](https://github.com/theia-ide/theia) using the [GLSP theia integration](https://github.com/eclipse-glsp/glsp-theia-integration).
 
 For more information, please visit the [Eclipse GLSP Umbrella repository](https://github.com/eclipse-glsp/glsp) and the [Eclipse GLSP Website](https://www.eclipse.org/glsp/). If you have questions, contact us on our [spectrum chat](https://spectrum.chat/glsp/) and have a look at our [communication and support options](https://www.eclipse.org/glsp/contact/).
```
Comment (martin-fleck-at) [inline README.md:28]: There should probably be another headline as this "more information"  section is not really part of "Structure".

**Review by martin-fleck-at — CHANGES_REQUESTED** [review]
Thank you very much for your change Tobias. I have a few remarks and I didn't manage to get the minimal example to run on my machine. Maybe you could provide some additional details?

Furthermore the commit message indicates that the issue is resolved. However, in the issue also an update of the animation GIF was requested. Probably one GIF per example?

File: Jenkinsfile (hunk 4 above)
Comment (tortmayr) [reply Jenkinsfile:89]: As discussed here, we currently are not planning on providing nightly deployment for the example repository.  Only releases are going to be deployed to npm/maven and this has to be done manually anyways. (At least for now) So we don need these build jobs anymore

File: README.md (hunk 5 above)
Comment (tortmayr) [reply README.md:20]: There is an open issue for adding the workflow example to this repository (https://github.com/eclipse-glsp/glsp/issues/138). So there will be another example soon :wink:

File: README.md (hunk 6 above)
Comment (tortmayr) [reply README.md:22]: Snapshot-glsp is the exectuable shaded maven bundle (fat-jar) that contains all dependencies and can be used to start the server via commandline.

I'm planning on providing additional details for running/building the examples soon. However, I'm planning on simplifying the start/build process  (similar to how it was done in the coffee editor) and once this is done I will update the README accordingly. For now I just wanted to make sure that the main README is (at least sort-of) up-to-date.

**Comment by tortmayr** [comment]
@martin-fleck-at  I updated the instructions for building & running the minimal example

**Review by martin-fleck-at — APPROVED** [review]
Change looks good to me now and I can execute the minimal example, thank you very much Tobias!

In the minimal example I get the following warning:
```
root WARN 2:59:06 PM minimal-diagram_0 GLSPActionDispatcher: Missing handler for action { actionKinds:
   [ 'createNode',
     'resolveNavigationTarget',
     'requestMarkers',
     'requestContextActions',
     'initializeClientSession',
     'reconnectEdge',
     'applyLabelEdit',
     'disposeClientSession',
     'computedBounds',
     'requestPopupModel',
     'changeBounds',
     'cut',
     'requestClipboardData',
     'paste',
     'glspUndo',
     'saveModel',
     'requestTypeHints',
     'glspRedo',
     'compound',
     'requestEditValidation',
     'requestNavigationTargets',
     'layout',
     'requestModel',
     'changeRoutingPoints',
     'deleteElement' ],
  kind: 'configureServerHandlers',
  __receivedFromServer: true }
```
even though I can execute undo/redo, saving, creating nodes, etc. Do you know if this is normal?

In any case, I don't think this should block us to merge this change but maybe we need to create a follow-up issue for that if there is no good explanation.

**Pull request merged**